File: apps/web/src/types/library.ts
export enum BookStatus {
    AVAILABLE = "AVAILABLE",
    ISSUED = "ISSUED",
    LOST = "LOST",
    DAMAGED = "DAMAGED"
}

export interface Book {
    id: number;
    title: string;
    author: string;
    isbn: string;
    publisher?: string;
    category?: string;
    location?: string;
    total_copies: number;
    available_copies: number;
    status: BookStatus;
}

export interface BookIssue {
    id: number;
    book_id: number;
    student_id: number;
    issue_date: string;
    due_date: string;
    return_date?: string;
    is_returned: boolean;
}

export interface LibraryFine {
    id: number;
    book_issue_id: number;
    amount: number;
    is_paid: boolean;
    payment_date?: string;
    remarks?: string;
}

File: apps/api/app/schemas/library.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import date, datetime
from app.models.library import BookStatus

# --- Book Schemas ---
class BookBase(BaseModel):
    title: str
    author: str
    isbn: str
    publisher: Optional[str] = None
    category: Optional[str] = None
    location: Optional[str] = None
    total_copies: int = 1

class BookCreate(BookBase):
    pass

class BookRead(BookBase):
    id: int
    available_copies: int
    status: BookStatus

class BookUpdate(BaseModel):
    title: Optional[str] = None
    author: Optional[str] = None
    publisher: Optional[str] = None
    category: Optional[str] = None
    location: Optional[str] = None
    total_copies: Optional[int] = None
    status: Optional[BookStatus] = None

# --- Issue Schemas ---
class BookIssueBase(BaseModel):
    book_id: int
    student_id: int
    due_date: date

class BookIssueCreate(BookIssueBase):
    pass

class BookIssueRead(BookIssueBase):
    id: int
    issue_date: date
    return_date: Optional[date] = None
    is_returned: bool

# --- Fine Schemas ---
class LibraryFineRead(BaseModel):
    id: int
    book_issue_id: int
    amount: float
    is_paid: bool
    payment_date: Optional[datetime] = None
    remarks: Optional[str] = None
